qutip/models/piqs.py

```diff
@@ -0,0 +1,1198 @@
+
+    Returns
+    -------
+    Nj: int
+        The number of Dicke ladders.
+    """
+    return _num_dicke_ladders(N)
+
+def num_tls(nds):
+    """Calculate the number of two-level systems.
+
+    Parameters
+    ----------
+    nds: int
+         The number of Dicke states.
+
+    Returns
+    -------
+    N: int
+        The number of two-level systems.
+    """
+    if np.sqrt(nds).is_integer():
+        # N is even
+        N = 2*(np.sqrt(nds)-1)
+    else:
+        # N is odd
+        N = 2*(np.sqrt(nds + 1/4)-1)
+    return int(N)
+
+
+class Dicke(object):
+    """The Dicke class which builds the Lindbladian and Liouvillian matrix.
+
+    Example
+    -------
+    >> from qutip.models.piqs import Dicke, j_algebra
+    >> N = 2
+    >> jx, jy, jz, jp, jm = j_algebra(N)
+    >> ensemble = Dicke(N, emission=1.)
+    >> L = ensemble.liouvillian()
+
+    Parameters
+    ----------
+    N: int
+        The number of two-level systems.
+
+    hamiltonian: :class: qutip.Qobj
+        A Hamiltonian in the reduced Dicke basis set.
+
+        The matrix dimensions are (nds, nds), with nds being the number of
+        dicke states. The hamiltonian can be built with the operators given
+        by the `j_algebra` function in the "dicke" basis.
+
+    emission: float
+        Incoherent emission coefficient (also nonradiative emission).
+        default: 0.0
+
+    dephasing: float
+        Local dephasing coefficient.
+        default: 0.0
+
+    pumping: float
+        Incoherent pumping coefficient.
+        default: 0.0
+
+    collective_emission: float
+        Collective (superradiant) emmission coefficient.
+        default: 0.0
+
+    collective_pumping: float
+        Collective pumping coefficient.
+        default: 0.0
+
+    collective_dephasing: float
+        Collective dephasing coefficient.
+        default: 0.0
+
+    Attributes
+    ----------
+    N: int
+        The number of two-level systems.
+
+    hamiltonian: :class: qutip.Qobj
+        A Hamiltonian in the reduced Dicke basis set.
+
+        The matrix dimensions are (nds, nds), with nds being the number of
+        dicke states. The hamiltonian can be built with the operators given
+        by the `j_algebra` function in the "dicke" basis.
+
+    emission: float
+        Incoherent emission coefficient (also nonradiative emission).
+        default: 0.0
+
+    dephasing: float
+        Local dephasing coefficient.
+        default: 0.0
+
+    pumping: float
+        Incoherent pumping coefficient.
+        default: 0.0
+
+    collective_emission: float
+        Collective (superradiant) emmission coefficient.
+        default: 0.0
+
+    collective_pumping: float
+        Collective pumping coefficient.
+        default: 0.0
+
+    collective_dephasing: float
+        Collective dephasing coefficient.
+        default: 0.0
+        
+    nds: int
+        The number of Dicke states.
+
+    dshape: tuple
+        The tuple (nds, nds).
+
+    blocks : list
```

Comment:
This blocks attribute does not appear to exist. Is it missing or should the docstring be removed?